index.js

```diff
@@ -284,12 +312,17 @@ function session(options) {
           chunk = !Buffer.isBuffer(chunk)
             ? Buffer.from(chunk, encoding)
             : chunk;
+          endArgs[0] = chunk;
```

Comment:
Since we pass `endArgs` to the original `_end()`, we need to keep it up to date. I believe the `chunk` is reassigned here because Node 0.8's `end()` only accepts a `Buffer`?

Mutating a function's `arguments` feels pretty bad, but it's by far the most elegant solution I came up with.

I'm not sure if this warrants a comment in the code.